samples/client/petstore/csharp/OpenAPIClient-generichost-net6.0-nrt/src/Org.OpenAPITools/Model/MixedPropertiesAndAdditionalPropertiesClass.cs

```diff
@@ -104,10 +104,13 @@ public override string ToString()
         IEnumerable<System.ComponentModel.DataAnnotations.ValidationResult> IValidatableObject.Validate(ValidationContext validationContext)
         {
             // UuidWithPattern (Guid) pattern
-            Regex regexUuidWithPattern = new Regex(@"[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}", RegexOptions.CultureInvariant);
-            if (false == regexUuidWithPattern.Match(this.UuidWithPattern.ToString()).Success)
+            if (this.UuidWithPattern != null)
```

Comment:
This produces a warning:
The result of the expression is always 'true' since a value of type 'Guid' is never equal to 'null' of type 'Guid?' 

You can fix it like this
```
{{#vendorExtensions.x-is-value-type}}
{{#isNullable}}
/* do a null check */
{{/isNullable}}
{{/vendorExtensions.x-is-value-type}}
{{^vendorExtensions.x-is-value-type}}
/* do a null check */
{{/vendorExtensions.x-is-value-type}}
```

Reply:
The solution seems complicated, is it warning worth it? Maybe it would be easier just to disable the warning instead? It does no harm, I guess?

Like this:
```
#pragma warning disable CS0472 // structs are never null
if (this.{{{name}}} != null)
{
…
}
#pragma warning restore CS0472
```

Going back to the proposed fix: 
- why there is a section with `do a null check` twice? 
- also, how do I insert a `do a null check` conditionally at the beginning and at the end of a code block, which as of now looks like this:

```
            if (this.{{{name}}} != null)
            {
                var regex{{{name}}} = new Regex(@"{{{vend[…✂…]
                if (!regex{{{name}}}.Match(this.{{{name}}}{{#isUuid}}.ToString(){{/isUuid}}).Success)
                {
                    yield return new System.ComponentModel.DataAnnotations.Validatio[…✂…]
                }
            }
```
Also, depending on the outer «if», the indentation of the inner one should be different. How do I do that? It is a lot of code to duplicate if I was to copy'n paste the inner «if» twice.

Reply:
Lets not leave the warning. I don't think ignoring the warning is appropriate either since we have the tools and knowledge to avoid the need. I care very much how the generated code looks, but I think it is okay if the templates are a bit ugly since no one should be looking at them.

```
{{#vendorExtensions.x-is-value-type}}
{{#isNullable}}
            if (this.{{{name}}} != null)
            {
                var regex{{{name}}} = new Regex(@"{{{vend[…✂…]
                if (!regex{{{name}}}.Match(this.{{{name}}}{{#isUuid}}.ToString(){{/isUuid}}).Success)
                {
                    yield return new System.ComponentModel.DataAnnotations.Validatio[…✂…]
                }
            }
{{/isNullable}}
{{^isNullable}}
                var regex{{{name}}} = new Regex(@"{{{vend[…✂…]
                if (!regex{{{name}}}.Match(this.{{{name}}}{{#isUuid}}.ToString(){{/isUuid}}).Success)
                {
                    yield return new System.ComponentModel.DataAnnotations.Validatio[…✂…]
                }
{{/isNullable}}
{{/vendorExtensions.x-is-value-type}}
{{^vendorExtensions.x-is-value-type}}
            if (this.{{{name}}} != null)
            {
                var regex{{{name}}} = new Regex(@"{{{vend[…✂…]
                if (!regex{{{name}}}.Match(this.{{{name}}}{{#isUuid}}.ToString(){{/isUuid}}).Success)
                {
                    yield return new System.ComponentModel.DataAnnotations.Validatio[…✂…]
                }
            }
{{/vendorExtensions.x-is-value-type}}
```

If you really don't want to repeat the logic inside the if statement, you can move it to a file and do this

{{#vendorExtensions.x-is-value-type}}
{{#isNullable}}
if (...){
    {{>FileName}}
}
{{/isNullable}}
{{^isNullable}}
{{>fileName}}
{{/isNullable}}
{{/vendorExtensions.x-is-value-type}}
{{^vendorExtensions.x-is-value-type}}
if (...) {
    {{>FileName}}
}
{{/vendorExtensions.x-is-value-type}}

but then you still have to handle the indentation. There are lambdas to indent x times. If you want to use it, you will have to add it to the AbstractCSharp addMustacheLambdas method. Here is the lambda `modules\openapi-generator\src\main\java\org\openapitools\codegen\templating\mustache\IndentedLambda.java`

Reply:
`why there is a section with do a null check twice?`

Because one is for reference types, the other is for nullable value types.

Reply:
> I care very much how the generated code looks, but I think it is okay if the templates are a bit ugly since no one should be looking at them.

This is so against my own principles in software development. The templates are the building blocks, lifeblood, crucial ingredient of this very tool. If they were ugly, bloated or unmaintainable, then the whole `csharp` plugin becomes worse, less "agile", harder to grasp and it would take more time and effort to develop it further.

On the other side we have the generated code, which is most of the time completely invisible. Nobody "maintains" the generated code. Most of the times it does not go into a code repository, because it's just a mean to an end. My understanding is that it could be a CLR bytecode as well and it would still serve its role.

So, we are going to sacrifice the essence of the library here, for no apparent reason in my opinion.

Could you, please:
- allow me to leave the pull request as is (and accept it), or
- allow me to add the `pragma warning disable` so there was no warning while compiling the generated code, or
- to implement the workaround yourself, so I could maintain my own integrity.

**I mean no disrespect.** I do acknowledge that different people have different priorities and point of views. We might disagree on this very issue and still have much in common in other matters.